scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py

```diff
@@ -339,7 +339,7 @@ def _minimize_trustregion_constr(fun, x0, args, grad,
 
     # Prepare constraints.
     prepared_constraints = [
-        PreparedConstraint(c, x0, sparse_jacobian, finite_diff_bounds)
+        PreparedConstraint(c, x0, sparse_jacobian, (-np.inf, np.inf))
```

Comment:
Not entirely sure about the modification in this line. Why not pass `finite_diff_bounds` to the PreparedConstraints?

This would not result in the finite difference method not respecting the constraints when applying disturbances to compute the derivative?

Reply:
Yes, this line looks fishy

Reply:
I would have to step through things again to jog my memory in case there was a deeper reason, but I think since we are catching any function evaluations outside of the bounds and returning inf, we will just shrink the trust region until we end up back in the feasible region. However, I think we can avoid this if we take a different approach and don't just catch any attempted function evaluations outside the feasible region (as below).